x/tx/aminojson/json_marshal.go

```diff
@@ -0,0 +1,357 @@
+package aminojson
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/iancoleman/strcase"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	cosmos_proto "github.com/cosmos/cosmos-proto"
+
+	"cosmossdk.io/api/amino"
+)
+
+// MessageEncoder is a function that can encode a protobuf protoreflect.Message to JSON.
+type MessageEncoder func(protoreflect.Message, io.Writer) error
+
+// FieldEncoder is a function that can encode a protobuf protoreflect.Value to JSON.
+type FieldEncoder func(AminoJSON, protoreflect.Value, io.Writer) error
+
+// AminoJSON is a JSON encoder that uses the Amino JSON encoding rules.
+type AminoJSON struct {
+	// maps cosmos_proto.scalar -> zero value factory
+	scalarEncoders  map[string]FieldEncoder
+	messageEncoders map[string]MessageEncoder
+	fieldEncoders   map[string]FieldEncoder
+}
+
+// NewAminoJSON returns a new AminoJSON encoder.
+func NewAminoJSON() AminoJSON {
+	aj := AminoJSON{
+		scalarEncoders: map[string]FieldEncoder{
+			"cosmos.Dec": cosmosDecEncoder,
+			"cosmos.Int": cosmosDecEncoder,
+		},
+		messageEncoders: map[string]MessageEncoder{
+			"key_field":      keyFieldEncoder,
+			"module_account": moduleAccountEncoder,
+		},
+		fieldEncoders: map[string]FieldEncoder{
+			"empty_string":     emptyStringEncoder,
+			"json_default":     jsonDefaultEncoder,
```

Comment:
What does this encoder do?

Reply:
Nothing, I think they were in place for testing, I forget now. Removed them.